rmgpy/cantherm/sensitivity.py

```diff
@@ -207,3 +209,176 @@ def plot(self):
+    `conditions`        A list of the conditions (each entry is a list of one T and one P quantities) at which the
+                        sensitivity coefficients are calculated
+    `job`               The PressureDependenceJob object
+    `rates`             A dictionary with netReactions as keys. Values are lists of forward rates from `job` for the
+                        respective path reaction at the respective `conditions` in the appropriate units
+    `sa_rates`          A dictionary with string representations of netReactions as keys. Values are dictionaries with
+                        Wells or TransitionStates as keys and each value is a list of forward rates from `job` at the
+                        respective `conditions` after perturbing the corresponding well or TS's E0
+    `sa_coefficients`   A dictionary with similar structure as `sa_rates`, containing the sensitivity coefficients
+                        in the forward direction
+    =================== ================================================================================================
+    """
+
+    def __init__(self, job, output_directory, perturbation):
+        self.job = job
+        self.output_directory = output_directory
+        self.conditions = self.job.sensitivity_conditions
+        self.rates = {}
+        for rxn in self.job.network.netReactions:
+            self.rates[str(rxn)] = []
+            for condition in self.conditions:
+                self.rates[str(rxn)].append(rxn.kinetics.getRateCoefficient(condition[0].value_si,
+                                                                            condition[1].value_si))
+        self.sa_rates = {}
+        self.sa_coefficients = {}
+        for rxn in self.job.network.netReactions:
+            self.sa_rates[str(rxn)] = {}
+            self.sa_coefficients[str(rxn)] = {}
+        self.perturbation = quantity.Quantity(perturbation, 'kcal/mol')
+        self.execute()
+
+    def execute(self):
+        """
+        Execute the sensitivity analysis for a :class:PressureDependenceJob: object
+        """
+        wells = []
+        wells.extend(self.job.network.reactants)
+        wells.extend(self.job.network.isomers)
+        wells.extend(self.job.network.products)
+        transition_states = []
+        for rxn in self.job.network.pathReactions:
+            # if rxn.transitionState is not None:
+            transition_states.append(rxn.transitionState)
+
+        for entry in wells + transition_states:
+            if entry in wells:
+                logging.info("\n\nPerturbing well '{0}' by {1}:".format(entry, self.perturbation))
+            else:
+                logging.info("\n\nPerturbing TS '{0}' by {1}:".format(entry.label, self.perturbation))
+            self.perturb(entry)
+            self.job.execute(outputFile=None, plot=False, print_summary=False)  # run the perturbed job
+            self.unperturb(entry)
+            if isinstance(entry, TransitionState):
```

Comment:
Is this done because E0 always needs to be positive to prevent negative values near zero?

Reply:
I removed this part, so now we use semi-normalized coefficients, and E0 isn't needed